src/lib/connectAxesToLayout.js

```diff
@@ -35,18 +33,41 @@ export default function connectAxesToLayout(WrappedComponent) {
       this.setLocals(nextProps, nextState, nextContext);
     }
 
-    // This function should be optimized. We can compare a list of
-    // axesNames to nextAxesNames and check gd.layout[axesN] for shallow
-    // equality. Unfortunately we are currently mutating gd.layout so the
-    // shallow check is not possible.
     setLocals(nextProps, nextState, nextContext) {
-      const {plotly, graphDiv, container, fullContainer} = nextContext;
+      const {container, fullContainer} = nextContext;
       const {axesTarget} = nextState;
-      if (plotly) {
-        this.axes = plotly.Axes.list(graphDiv);
-      } else {
-        this.axes = [];
-      }
+
+      this.axes = [];
+
+      // Plotly.js should really have a helper function for this, but until it does..
+      Object.keys(fullContainer._subplots)
```

Comment:
ok, so subplots will be step 2, after this pr, that is why I disabled a test somewhat related to them, above.
Need to think about this a little more when I'm writing this function here.